src/App.tsx

```diff
@@ -16,20 +35,23 @@ export class App extends React.Component<{}, State> {
             <div className="field">
               <label htmlFor="search-query" className="label">
                 Search movie
-              </label>
 
-              <div className="control">
-                <input
-                  type="text"
-                  id="search-query"
-                  className="input"
-                  placeholder="Type search word"
-                />
-              </div>
+                <div className="control">
+                  <input
+                    type="text"
+                    id="search-query"
+                    className="input"
+                    placeholder="Type search word"
+                    name="query"
+                    value={this.state.query}
+                    onChange={this.handleChange}
+                  />
+                </div>
+              </label>
             </div>
           </div>
 
-          <MoviesList movies={moviesFromServer} />
+          <MoviesList movies={this.visibleMovies()} />
```

Comment:
not do like this, pass to props already calculated movies (call this method in render)